### ISSUE TITLE
ncurses error when using binary

### ISSUE THREAD

**Issue by faddat** [mimblewimble/grin#2106]
This release:

https://github.com/mimblewimble/grin/releases/tag/v0.4.2

When using the v0.4.2 binary, I get the following ncurses error:

./grin: error while loading shared libraries: libncursesw.so.5: cannot open shared object file: No such file or directory

I don't get this error when building using cargo from source code.  When I build it myself Grin works just fine :).  Arch Linux.  Happy to investigate further.

**Comment by quentinlesceller** [comment]
Hey @faddat, just curious if you tried to install `ncurses` on your Arch linux?

**Comment by tsusanka** [comment]
I believe this is because Arch Linux has already the newer version of libncursesw.so - libncursesw.so.6:

```
$ ls /usr/lib/libncursesw.so.6
/usr/lib/libncursesw.so.6
$ ls /usr/lib/libncursesw.so.5
ls: cannot access '/usr/lib/libncursesw.so.5': No such file or directory
```

Creating a simlink 6 -> 5 works just fine:

```
ln -s /usr/lib/libtinfo.so.6 /usr/lib/libtinfo.so.5
``` 

and exactly the same goes for /usr/lib/libtinfo.so. Grin lunches just fine after these fixes

**Comment by Lazydayz137** [comment]
I installed 5 on a vultr server running Ubuntu 18.xx, still get the error.  Not paying attention thought I might need the newer 6.1, so compiled and installed.  Same error of course and the simlink back to 5 still gives initial error.  Going to try rolling back to good old 16.xx I think.

**Comment by Lazydayz137** [comment]
Works like a charm, no ncurses install required(5 already there) on a Vultr Ubuntu 16.04 clean install w/upgrades.  Just as an update to my previous comment.  Doesn't change OP issue.

**Comment by jeroen7s** [comment]
same issue here
using Manjaro (arch based)

**Comment by jasonzhouu** [comment]
fixed this problem in ubuntu 18.10:
```bash
cd /lib/x86_64-linux-gnu
ln -s libncursesw.so.6 libncursesw.so.5
ln -s libtinfo.so.6 libtinfo.so.5
```

i don't come across this problem in ubuntu 18.04, and these two is appreaded in version 5.

**Comment by jeroen7s** [comment]
cding into /usr/lib/ and using your symlink commands from there seems to solve the issue on manjaro (and probably Arch as well)

**Comment by quentinlesceller** [comment]
Looks like the a distro related error and not a grin error itself. Closing.

**Comment by jeroen7s** [comment]
Seems to me like an issue with grin not recognizing libncurse version 6, only version 5
Once ubuntu updates to v6 this issue will also occur there

**Comment by jeroen7s** [comment]
Seems like it already occurs in version 18.10 reading the comments above

**Comment by cool007zqw** [comment]
> 
> 
> I believe this is because Arch Linux has already the newer version of libncursesw.so - libncursesw.so.6:
> 
> ```
> $ ls /usr/lib/libncursesw.so.6
> /usr/lib/libncursesw.so.6
> $ ls /usr/lib/libncursesw.so.5
> ls: cannot access '/usr/lib/libncursesw.so.5': No such file or directory
> ```
> 
> Creating a simlink 6 -> 5 works just fine:
> 
> ```
> ln -s /usr/lib/libtinfo.so.6 /usr/lib/libtinfo.so.5
> ```
> 
> and exactly the same goes for /usr/lib/libtinfo.so. Grin lunches just fine after these fixes

After making the link, it comes with another error:

`Segmentation fault (core dumped)`

**Comment by jeroen7s** [comment]
@cool007zqw try the code from jasonzhouu, that fixed the errors for me on Manjaro

**Comment by mjssln** [comment]
> fixed this problem in ubuntu 18.10, grin 1.0.1:
> 
> ```shell
> cd /lib/x86_64-linux-gnu
> ln -s libncursesw.so.6 libncursesw.so.5
> ln -s libtinfo.so.6 libtinfo.so.5
> ```
> 
> in ubuntu 18.04, this problem don't exist, and these two file is appeared in version 5.

Ubuntu 19.04
Fix for me ! 
Thx

**Comment by FTLIan** [comment]
I am on Ubuntu 19.04 and the symlinks have not solved the problem for me.  After running the above commands, trying to run grin gives me these errors:

./grin: /lib/x86_64-linux-gnu/libtinfo.so.5: version NCURSES_TINFO_5.0.19991023' not found (required by ./grin)
./grin: /lib/x86_64-linux-gnu/libncursesw.so.5: versionNCURSESW_5.6.20061217' not found (required by ./grin)
./grin: /lib/x86_64-linux-gnu/libncursesw.so.5: version `NCURSESW_5.1.20000708' not found (required by ./grin)

**Comment by quentinlesceller** [comment]
Hi @FTLIan, I will spin up a 19.04 VM and see if I can help you.

**Comment by quentinlesceller** [comment]
Hi @FTLIan unfortunately I was not able to run the binaries on Ubuntu 19.04. I opened the following issue https://github.com/mimblewimble/grin/issues/3022.

**Comment by slicken** [comment]
I am on Ubuntu 19.10 and have same issue as FTLIan

slicken@slk:~/grin$ ./grin 
./grin: error while loading shared libraries: libncursesw.so.5: cannot open shared object file: No such file or directory

slicken@slk:/lib/x86_64-linux-gnu$ sudo ln -s libncursesw.so.6 libncursesw.so.5
slicken@slk:/lib/x86_64-linux-gnu$ sudo ln -s libtinfo.so.6 libtinfo.so.5

slicken@slk:~/grin$ ./grin 
./grin: /lib/x86_64-linux-gnu/libtinfo.so.5: version `NCURSES_TINFO_5.0.19991023' not found (required by ./grin)
./grin: /lib/x86_64-linux-gnu/libncursesw.so.5: version `NCURSESW_5.6.20061217' not found (required by ./grin)
./grin: /lib/x86_64-linux-gnu/libncursesw.so.5: version `NCURSESW_5.1.20000708' not found (required by ./grin)


what should I do?

**Comment by Sky-Bly** [comment]
Resolves:

sudo apt-get install -y libncursesw5

**Comment by lehnberg** [comment]
@quentinlesceller I just encountered this issue as well (Debian Buster) with newer versions of libncurses already installed on the system.

@Sky-Bly's fix worked for me (thanks for that)
>  sudo apt-get install -y libncursesw5

...but I would tend to agree with @jeroen7s's comment above:
> Seems to me like an issue with grin not recognizing libncurse version 6, only version 5

Would be nice if we could get this addressed somehow, I don't think it's obvious for end users that they need to have a specific older version of ncurses installed in order for the binary to work. Perhaps makes sense to reopen?

**Comment by slicken** [comment]
I agree, fixing small things like this matter
easy to install/use, with no hurdles is good for adoption

**Comment by quentinlesceller** [comment]
I'll take a look.

**Comment by quentinlesceller** [comment]
To clarify: the issue is with the linking of libncursesw-6 which is the wide character library. This looks similar to https://github.com/jeaye/ncurses-rs/issues/151. 
However since we are using cursive might be something related to https://github.com/gyscos/cursive/issues/411. We are however using an old version of Cursive 0.12 instead of 0.14.

**Comment by quentinlesceller** [comment]
Since libncursesw5 and libncursesw6 uses the same API. Doing an
```
apt install libncurses5
```

on Ubuntu solved it (tested on Ubuntu 20.04). I expect it to be the same on other distros.

**Comment by chillmetin** [comment]
> fixed this problem in ubuntu 18.10, grin 1.0.1:
> 
> ```shell
> cd /lib/x86_64-linux-gnu
> ln -s libncursesw.so.6 libncursesw.so.5
> ln -s libtinfo.so.6 libtinfo.so.5
> ```
> 
> in ubuntu 18.04, this problem don't exist, and these two file is appeared in version 5.

fixed the issue in pop!_OS 22.04
